services/garden/garden.selectors.ts

```diff
@@ -0,0 +1,110 @@
+import { createSelector } from "@reduxjs/toolkit";
```

Comment:
@Uken81 selectors have been moved here, they've been broken out of the object they were in so they can reference each other which has cleaned up a lot of code, but then there's a more added with createSelector. 

It's good to get the memorization benefits but it doesn't handle multiple arguments to the selector well. Mainly the type but not the name of the arguments aren't in IntelliSense and you have to look at the selector to see what it needs.